File: settlement/room/military/artillery/Hoverer.java
package settlement.room.military.artillery;

import init.sprite.SPRITES;
import settlement.thing.projectiles.Trajectory;
import util.gui.misc.GBox;
import util.info.GFORMAT;
import util.text.Dic;

class Hoverer {

	public static void hover(GBox box, ArtilleryInstance i) {
		ArtilleryInstance ins = (ArtilleryInstance) i;
		
		if (ins.mustered()) {
			box.textL(Dic.¤¤Musterd);
			box.NL();
			if (i.isLoaded)
				box.textL(Dic.¤¤ReadyFire);
			else {
				box.textL(Dic.¤¤Reloading);
				box.tab(5);
				box.add(GFORMAT.perc(box.text(), i.progress()));
			}
				box.NL(4);
		}
		
		if (i.hasTrajectory && ins.mustered()) {
			box.textL(Dic.¤¤Attacking);
		}
		
		box.add(SPRITES.icons().s.human);
		box.add(GFORMAT.iofkInv(box.text(), ins.men, 6));
		box.NL(6);
		box.sep();
		i.blueprintI().projectile.hover(box, i.blueprintI().info.name, i.blueprintI().ref()*i.getDegrade(), Trajectory.RELEASE_HEIGHT);
		
	}
	
}

File: LegacySource/settlement/room/military/artillery/Hoverer.cs
using System;
using settlement.room.military.artillery;
using init.sprite;
using settlement.thing.projectiles;
using util.gui.misc;
using util.info;
using util.text;

class Hoverer
{
    public static void Hover(GBox box, ArtilleryInstance i)
    {
        ArtilleryInstance ins = (ArtilleryInstance)i;

        if (ins.Mustered())
        {
            box.TextL(Dic.¤¤Musterd);
            box.NL();
            if (i.IsLoaded)
                box.TextL(Dic.¤¤ReadyFire);
            else
            {
                box.TextL(Dic.¤¤Reloading);
                box.Tab(5);
                box.Add(GFORMAT.Perc(box.Text(), i.Progress()));
            }
            box.NL(4);
        }

        if (i.HasTrajectory && ins.Mustered())
        {
            box.TextL(Dic.¤¤Attacking);
        }

        box.Add(SPRITES.Icons().S.Human);
        box.Add(GFORMAT.IofkInv(box.Text(), ins.Men, 6));
        box.NL(6);
        box.Sep();
        i.BlueprintI().Projectile.Hover(box, i.BlueprintI().Info.Name, i.BlueprintI().Ref() * i.GetDegrade(), Trajectory.RELEASE_HEIGHT);
    }
}